### PR TITLE
Add JDK annotations for the Side Effects Only Checker

### PR DESCRIPTION
**Pull request by jyoo980**

### CODE REVIEW

File: src/java.base/share/classes/java/util/Iterator.java [hunk 1]
```diff
@@ -83,6 +84,7 @@ public interface Iterator<E> {
      * @return the next element in the iteration
      * @throws NoSuchElementException if the iteration has no more elements
      */
+    @SideEffectsOnly("this")
```
Comment (jyoo980) [inline src/java.base/share/classes/java/util/Iterator.java:90]: This actually might eliminate the error in the [`SetIterator.java`](https://github.com/typetools/checker-framework/blob/da7819dd8e2125d8c040c66e8f1b6c0d7e6a10f3/checker/tests/nullness/SetIteratorTest.java#L29) test case.

I'm not sure why the following error should be issued:

```java
    if (edges.get(parentNode) != null) {
      for (String childNode : edges.get(parentNode).keySet()) {
        // :: error: (dereference.of.nullable)
        edges.get(parentNode).toString();
        for (String childNodeEdgeX : edges.get(parentNode).get(childNode)) {
          childrenString += " " + childNode + "(" + childNodeEdgeX + ")";
        }
      }
    }
```
The loop executes only if `edges.get(parentNode)` is non-null; I don't see how the enhanced-for can make this fact false, and so the call to `edges.get(parentNode)` should be legal.

**Comment by mernst** [comment]
CI is failing for plume-util.  Could you make a branch of it (in your fork) named exactly yoo/side-effects-only, fix the problem, pull-request it, and then link that pull request to this one?

Also, this pull request seems premature -- pulling occurrences of `@SideEffectsOnly` into master before the master branch of the Checker Framework supports it.  Am I mistaken about that?

**Review by mernst — APPROVED** [review]

**Comment by mernst** [comment]
@smillst I think it is acceptable to merge this even before merging the Side Effect Free Checker.  What do you think?

**Comment by smillst** [comment]
> @smillst I think it is acceptable to merge this even before merging the Side Effect Free Checker. What do you think?

I'm fine with that.  Though, if we need to update the qualifiers in the jdk we will have to use the Side Effect Free Checker branch.

**Review by smillst — APPROVED** [review]

**Pull request merged**